builder/builder-next/adapters/containerimage/pull.go

```diff
@@ -19,7 +19,7 @@ import (
 	ctdreference "github.com/containerd/containerd/reference"
 	"github.com/containerd/containerd/remotes"
 	"github.com/containerd/containerd/remotes/docker"
-	"github.com/containerd/containerd/remotes/docker/schema1"
+	"github.com/containerd/containerd/remotes/docker/schema1" //nolint:staticcheck // Ignore SA1019: "github.com/containerd/containerd/remotes/docker/schema1" is deprecated: use images formatted in Docker Image Manifest v2, Schema 2, or OCI Image Spec v1.
```

Comment:
probably also needs a squash with the containerd update

Reply:
https://github.com/moby/moby/actions/runs/6261172963/job/17000742211?pr=45966 caught another schema1-deprecation that needs lint-skipping.

```
daemon/containerd/image_pull.go:118:22: SA1019: containerd.WithSchema1Conversion is deprecated: use Schema 2 or OCI images. (staticcheck)
	opts = append(opts, containerd.WithSchema1Conversion)
	                    ^
```

I noticed #42300 doesn't appear to remove either of these though, unless I'm misunderstanding, and that's for something else.

Reply:
Thanks for spotting that!

Reply:
https://github.com/moby/moby/pull/42300 probably still needs some cleaning up; it started from a much larger branch, then chunks got "chipped off" from that and merged as separate PRs where we could.